src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.cpp

```diff
@@ -59,22 +67,28 @@ void EnemyFreeKickPlayFSM::setTactics(const Update& event, unsigned int num_tact
     std::vector<DefenderAssignment> crease_defender_assignments;
     std::vector<DefenderAssignment> pass_defender_assignments;
     std::queue<DefenderAssignment> assignments_skipped;
-    for (unsigned int i = 0; i < num_defenders; i++)
+    for (int i = 0; i < num_defenders; i++)
     {
         DefenderAssignment defender_assignment;
-        if (i + assignments_skipped.size() < assignments.size() - 1)
+        int assignment_index_with_skipped =
+            static_cast<int>(i + assignments_skipped.size());
+        int assignment_index_last_assignment = static_cast<int>(assignments.size() - 1);
```

Comment:
```suggestion
        int assignment_index_last_assignment = static_cast<int>(assignments.size()) - 1;
```
This doesn't work the way you'd expect. This can still overflow if the `assignments` vector is empty.